backend_addon/{{ cookiecutter.__folder_name }}/Makefile

```diff
@@ -45,77 +46,88 @@ all: build
 help: ## This help message
 	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
 
-$(BIN_FOLDER)/pip $(BIN_FOLDER)/tox $(BIN_FOLDER)/pipx $(BIN_FOLDER)/uv $(BIN_FOLDER)/mxdev:
-	@echo "$(GREEN)==> Setup Virtual Env$(RESET)"
-	$(PYTHON) -m venv $(VENV_FOLDER)
-	$(BIN_FOLDER)/pip install -U "pip" "uv" "wheel" "pipx" "tox" "pre-commit"
-	if [ -d $(GIT_FOLDER) ]; then $(BIN_FOLDER)/pre-commit install; else echo "$(RED) Not installing pre-commit$(RESET)";fi
+requirements-mxdev.txt: ## Generate constraints file
+	@echo "$(GREEN)==> Generate constraints file$(RESET)"
+	@echo '-c https://dist.plone.org/release/$(PLONE_VERSION)/constraints.txt' > requirements.txt
+	@uvx mxdev -c mx.ini
+
+$(VENV_FOLDER): requirements-mxdev.txt ## Install dependencies
+	@echo "$(GREEN)==> Install environment$(RESET)"
+	@uv venv $(VENV_FOLDER)
+	@uv pip install -r requirements-mxdev.txt
 
-instance/etc/zope.ini: $(BIN_FOLDER)/pip  ## Create instance configuration
+.PHONY: sync
+sync: $(VENV_FOLDER) ## Sync project dependencies
+	@echo "$(GREEN)==> Sync project dependencies$(RESET)"
+	@uv pip install -r requirements-mxdev.txt
+
+instance/etc/zope.ini instance/etc/zope.conf: ## Create instance configuration
 	@echo "$(GREEN)==> Create instance configuration$(RESET)"
-	$(BIN_FOLDER)/pipx run cookiecutter -f --no-input --config-file instance.yaml gh:plone/cookiecutter-zope-instance
+	@uvx cookiecutter -f --no-input --config-file instance.yaml gh:plone/cookiecutter-zope-instance
 
 .PHONY: config
 config: instance/etc/zope.ini
 
-.PHONY: build-dev
-build-dev: config ## Install Plone packages
-	@echo "$(GREEN)==> Setup Build$(RESET)"
-	$(BIN_FOLDER)/pipx run mxdev -c mx.ini
-	VIRTUAL_ENV=.venv $(BIN_FOLDER)/uv pip install -r requirements-mxdev.txt
-
 .PHONY: install
-install: build-dev ## Install Plone
+install: $(VENV_FOLDER) config ## Install Plone and dependencies
 
-.PHONY: build
-build: build-dev ## Install Plone
 
 .PHONY: clean
-clean: ## Clean environment
+clean: ## Clean installation and instance
 	@echo "$(RED)==> Cleaning environment and build$(RESET)"
-	rm -rf $(VENV_FOLDER) pyvenv.cfg .installed.cfg instance/etc .tox .venv .pytest_cache
+	@rm -rf $(VENV_FOLDER) pyvenv.cfg .installed.cfg instance .venv .pytest_cache .ruff_cache constraints* requirements*
```

Comment:
I think turning all constraints and requirements files into temporary files is going to catch some people by surprise. It would be better to use specific names for the files used with mxdev and only clean and gitignore those.

Reply:
As requirements.txt is generated from the Makefile, it makes no sense to keep it in the repository. I see your point about the confusion, but I would prefer to explain in docs than to handle the confusion of distinct Plone versions defined as env vars x the requirement file.